SSA/Core/MLIRSyntax/GenericParser.lean

```diff
@@ -692,15 +715,19 @@ macro "[mlir_attr_entry|" entry:mlir_attr_entry "]" : term => do
   `(AttrEntry.mk $(Lean.quote key) $value)
 
 declare_syntax_cat mlir_attr_dict
-syntax "{" sepBy(mlir_attr_entry, ",") "}" : mlir_attr_dict
+syntax  "<" ? "{" sepBy(mlir_attr_entry, ",") "}" ">" ?  : mlir_attr_dict
 syntax "[mlir_attr_dict|" mlir_attr_dict "]" : term
 
 macro_rules
-| `([mlir_attr_dict| {  $attrEntries,* } ]) => do
+| `([mlir_attr_dict| $[<%$caretLeft]? {  $attrEntries,* } $[>%$caretRight]? ]) => do
+        match caretLeft, caretRight with
+        | none, some y => Macro.throwErrorAt y "Caret closed without being opened"
+        | some x, none => Macro.throwErrorAt x "Caret opened without being closed"
+        | _ , _ =>
+
         let attrsList <- attrEntries.getElems.toList.mapM (fun x => `([mlir_attr_entry| $x]))
         let attrsList <- quoteMList attrsList (<- `(MLIR.AST.AttrEntry _))
         `(AttrDict.mk $attrsList)
```

Comment:
@AtticusKuhn, you should consider removing the whitespace line, and indenting the three lines below it